src/control/server/faultdomain.go

```diff
@@ -63,3 +73,24 @@ func newFaultDomainFromConfig(domainStr string) (*system.FaultDomain, error) {
 	}
 	return fd, nil
 }
+
+func getFaultDomainFromCallback(callbackPath string) (*system.FaultDomain, error) {
+	if callbackPath == "" {
+		return nil, errors.New("no callback path supplied")
+	}
+
+	// Fault callback can't be an arbitrary command. Must point to a
+	// specific executable file.
+	if err := unix.Stat(callbackPath, nil); os.IsNotExist(err) {
+		return nil, FaultConfigFaultCallbackNotFound
+	}
+
+	output, err := exec.Command(callbackPath).Output()
+	if os.IsPermission(err) {
+		return nil, FaultConfigFaultCallbackBadPerms
+	} else if err != nil {
+		return nil, FaultConfigFaultCallbackFailed
```

Comment:
Might be good to make FaultConfigFaultCallbackFailed a function that takes the error so that we can display it. Otherwise the admin might not know why the callback failed.

Reply:
Good idea. I'll add that.